Std/Data/String/Basic.lean

```diff
@@ -9,29 +9,52 @@ import Std.Data.Nat.Lemmas
 
 namespace Substring
 
+/-- Check if two substrings are equal as strings. -/
+def eqAsString (s t : Substring) : Bool :=
+  if s.bsize ≠ t.bsize then
+    false
+  else
+    go s.bsize 0
```

Comment:
`go s.bsize 0` is equivalent to `String.substrEq s.str s.startPos t.str t.startPos s.bsize`